src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/LdapAccessPolicyDAO.scala

```diff
@@ -107,15 +112,34 @@ class LdapAccessPolicyDAO(protected val ldapConnectionPool: LDAPConnectionPool,
     newPolicy
   }
 
-  override def listAccessPolicies(resourceTypeName: ResourceTypeName, userId: WorkbenchUserId): Future[Set[ResourceIdAndPolicyName]] = Future {
-    val entry = ldapConnectionPool.getEntry(userDn(userId), Attr.memberOf)
-    val groupDns = getAttributes(entry, Attr.memberOf).getOrElse(Set.empty).toSet
+  override def listAccessPolicies(resourceTypeName: ResourceTypeName, userId: WorkbenchUserId): IO[Set[ResourceIdAndPolicyName]] = {
+    val policyDnPattern = dnMatcher(Seq(Attr.policy, Attr.resourceId), resourceTypeDn(resourceTypeName))
+    for{
+      entry <- runBlocking(IO(ldapConnectionPool.getEntry(userDn(userId), Attr.memberOf)))
+    } yield {
+      val groupDns = Option(entry).flatMap(e => Option(getAttributes(e, Attr.memberOf))).getOrElse(Set.empty)
+      groupDns.collect{case policyDnPattern(policyName, resourceId) => ResourceIdAndPolicyName(ResourceId(resourceId), AccessPolicyName(policyName))}
+    }
+  }
 
-    val policyDnPattern = dnMatcher(List(Attr.policy, Attr.resourceId), resourceTypeDn(resourceTypeName))
+  override def listResourceWithAuthdomains(resourceTypeName: ResourceTypeName, resourceId: Set[ResourceId]): IO[Set[Resource]] = {
+    val filters = resourceId.grouped(batchSize).map(
+      batch =>
+        Filter.createORFilter(
+          batch.map(r => Filter.createANDFilter(Filter.createEqualityFilter(Attr.resourceId, r.value), Filter.createNOTFilter(Filter.createPresenceFilter(Attr.policy)))
+          ).asJava)).toSeq
+
+    for{
+      stream <- runBlocking(IO(ldapSearchStream(resourcesOu, SearchScope.SUB, filters:_*)(et => unmarshalResourceAuthDomain(et, resourceTypeName))))
+      res <- IO.fromEither(stream.toList.parSequence.leftMap(s => new WorkbenchException(s)))
```

Comment:
`toList` is needed since there's no `Paralell` for `Set`.  `parSequence` turns `List[Either[String, Resource]]` into `Either[String, List[Resource]` (the resulting `String` is concatenated value in `List[Either[String, Resource]]`). 

```
qi-sam@   val strErr1: Either[String, String] = "error 1. ".asLeft[String]
strErr1: Either[String, String] = Left("error 1. ")

qi-sam@   val strErr2: Either[String, String] = "error 2. ".asLeft[String]
strErr2: Either[String, String] = Left("error 2. ")

qi-sam@   val str3: Either[String, String] = "this is fine".asRight[String]
str3: Either[String, String] = Right("this is fine")

qi-sam@

qi-sam@   val result2: Either[String, List[String]] = List(strErr1, strErr2, str3).parSequence
result2: Either[String, List[String]] = Left("error 1. error 2. ")

```

The API is defined with returning a `Set`, so we're loading the whole stream anyway. If that's an issue, we'll probably need to do something else.